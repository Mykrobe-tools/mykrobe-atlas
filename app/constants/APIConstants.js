/* @flow */

export const BASE_URL: string =
  process.env.NODE_ENV === 'production'
    ? (process.env.API_URL: any)
    : 'https://api.atlas-dev.makeandship.com';

// export const BASE_URL = 'https://localhost:3001';
// export const BASE_URL = 'http://13.69.243.89:8000';

export const USER_COOKIE_NAME = 'api.atlas.makeandship.com';
