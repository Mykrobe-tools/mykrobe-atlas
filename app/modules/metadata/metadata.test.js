/* @flow */

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';

import { BASE_URL } from '../../constants/APIConstants.js';

import reducer, {
  initialState,
  postMetadataForm,
  setMetadata,
} from './metadata';

const createMockStore = configureMockStore([thunk]);

describe('metadata module', () => {
  const store = createMockStore(initialState);
  let mockState;

  beforeEach(() => {
    store.clearActions();
    nock.cleanAll();
  });

  it('should return the initial state', () => {
    mockState = reducer();
    expect(mockState).toEqual(initialState);
  });

  it('should handle "setMetadata" action', async () => {
    await store.dispatch(setMetadata({ lorem: 'ipsum' }));
    const dispatchedActions = store.getActions();
    mockState = reducer(undefined, dispatchedActions[0]);
    expect(mockState).toEqual({
      metadata: { lorem: 'ipsum' },
    });
  });

  it('should handle "postMetadataForm" action', async () => {
    nock(BASE_URL)
      .put('/experiments/1')
      .reply(200, { status: 'ok' });
    await store.dispatch(postMetadataForm('1', {}));
    const dispatchedActions = store.getActions();
    mockState = reducer(undefined, dispatchedActions[0]);
    expect(mockState).toMatchSnapshot();
  });
});
