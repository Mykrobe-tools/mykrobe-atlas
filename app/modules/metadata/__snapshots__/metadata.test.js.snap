// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metadata module should handle "postMetadataForm" action 1`] = `
Object {
  "metadata": Object {
    "date": "",
    "hivPositive": "",
    "labId": "",
    "location": "GB",
    "patientHistory": "",
    "patientId": "",
    "responsiblePersonData": "",
    "responsiblePersonId": "",
    "sampleId": "",
    "sampleType": "",
    "sequencingMachine": "",
    "shareSequence": true,
    "susceptibility": Object {},
    "treatedForTB": "",
  },
}
`;
